File: pages/base.page.ts
import { Page } from '@playwright/test';

export class BasePage {
  constructor(protected readonly page: Page) {}

  async goto(url: string) {
    await this.page.goto(url, { waitUntil: 'domcontentloaded' });
  }

  async waitForStableLayout(timeoutMs = 1000) {
    await this.page.waitForLoadState('load').catch(() => {});
    await this.page.waitForTimeout(timeoutMs);
  }
}

File: pages/base_page.py
from playwright.sync_api import Page


class BasePage:
    def __init__(self, page: Page):
        self.page = page

    def goto(self, url: str) -> None:
        self.page.goto(url, wait_until="domcontentloaded")

    def wait_for_stable_layout(self, timeout_ms: int = 1000) -> None:
        try:
            self.page.wait_for_load_state("load")
        except Exception:
            pass
        self.page.wait_for_timeout(timeout_ms)
